src/matrix/verification/SAS/stages/WaitForIncomingMessageStage.ts

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2023 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import {BaseSASVerificationStage, Options} from "./BaseSASVerificationStage";
+import {FragmentBoundaryEntry} from "../../../room/timeline/entries/FragmentBoundaryEntry.js";
+
+export class WaitForIncomingMessageStage extends BaseSASVerificationStage {
+    constructor(private messageType: string, options: Options) {
+        super(options);
+    }
+
+    async completeStage() {
+        await this.log.wrap("WaitForIncomingMessageStage.completeStage", async (log) => {
+            const content = await this.fetchMessageEventsFromTimeline();
+            console.log("content found", content);
+            this.nextStage.setResultFromPreviousStage({
+                ...this.previousResult,
+                [this.messageType]: content
+            });
+            this.dispose();
+        });
+    }
+
+    private fetchMessageEventsFromTimeline() {
+        // todo: add timeout after 10 mins
+        return new Promise(resolve => {
+            this.track(
+                this.room._timeline.entries.subscribe({
```

Comment:
Fwiw, `room._timeline` will only exist if the room is open in the UI. Either we should call `room.openTimeline()` here or we need to add some sort of callback for incoming timeline events like we added for state events for calls.